packages/react/src/components/Pagination/Pagination.stories.tsx

```diff
@@ -1,127 +1,100 @@
+import { useArgs } from '@storybook/preview-api';
 import type { Meta, StoryFn } from '@storybook/react';
-import { useEffect, useState } from 'react';
+import { useState } from 'react';
 
-import { Pagination, usePagination } from '.';
+import { Pagination, type UsePaginationProps, usePagination } from '.';
 
 export default {
   title: 'Komponenter/Pagination',
   component: Pagination,
 } as Meta;
 
-export const Preview: StoryFn<typeof Pagination> = (args) => {
-  const [currentPage, setCurrentPage] = useState(args.currentPage);
-
-  useEffect(() => {
-    setCurrentPage(args.currentPage);
-  }, [args.currentPage]);
-
-  return (
-    <Pagination {...args} onChange={setCurrentPage} currentPage={currentPage} />
-  );
-};
-Preview.args = {
-  'aria-label': 'Sidenavigering',
-  size: 'md',
-  nextLabel: 'Neste',
-  previousLabel: 'Forrige',
-  totalPages: 10,
-  hideLabels: false,
-  compact: false,
-  currentPage: 1,
-  itemLabel: (num) => `Side ${num}`,
-};
-
-export const UsePagination: StoryFn<typeof Pagination> = (args) => {
-  const { totalPages = 10 } = args;
-  const {
-    pages,
-    currentPage,
-    setCurrentPage,
-    nextPage,
-    previousPage,
-    showNextPage,
-    showPreviousPage,
-  } = usePagination({
-    currentPage: 4,
-    totalPages,
+export const Preview: StoryFn<UsePaginationProps> = (args) => {
```

Comment:
I can't use the controls here to change the `size`. 
See comment about context for a possible solution.

Reply:
Have discussed this with Micke, and will keep `size` on `Pagination` for now, as this matches the pattern of all other components, and is less error-prone than sniffing child components.